pkg/logbridge/logbridge.go

```diff
@@ -0,0 +1,88 @@
+package logbridge
+
+import (
+	"bufio"
+	"io"
+
+	"github.com/square/p2/pkg/logging"
+)
+
+// This type is agnostic of pipes. os.File conforms to io.Reader and io.Writer
+// so it should not be difficult to migrate
+type Bridge struct {
+	src  io.Reader
+	dest io.Writer
+
+	lines         chan []byte
+	needsTeardown bool
+}
+
+// New takes two files src and dest and returns a composite object of the
+// two.
+func New(src io.Reader, dest io.Writer, capacity int) Bridge {
+	if src == nil {
+		panic("Argument Error: Reader cannot be nil")
+	}
+	if dest == nil {
+		panic("Argument Error: Reader cannot be nil")
+	}
+
+	return Bridge{
+		src:           src,
+		dest:          dest,
+		lines:         make(chan []byte, capacity),
+		needsTeardown: true,
+	}
+}
+
+// Run is responsible for reading from src and writing to dst until src is
+// consumed or an error is encountered. The caller is responsible for closing
+// and flushing the reader and writer.
+func (b *Bridge) Run() error {
+	var buf []byte
+	var moreLines bool
+	go b.blockingRead()
+
+	for {
+		select {
+		case buf, moreLines = <-b.lines:
+			if !moreLines {
+				return nil
+			}
+			_, err := b.dest.Write(buf)
+			if err != nil {
+				logging.DefaultLogger.WithError(err).Error("logbridge: encountered fatal error during write")
```

Comment:
avoid default logger outside tests and `package main`. For example if I want to add custom fields to the logger, I'd rather not mutate the default logger global to do so.